### PR TITLE
Added functionality in esminiRMLib to see if a road_id is a successor…

### PR DESCRIPTION
**Pull request by seifbourogaa**
Added basic functionality in esminiRMLib to see if a road_id is a successor to another road_id.

### CODE REVIEW

File: EnvironmentSimulator/Libraries/esminiRMLib/esminiRMLib.cpp [hunk 1]
```diff
@@ -239,6 +239,28 @@ extern "C"
...
+            roadmanager::Road *successor_road = odrManager->GetRoadById(other_id);
+
+            if(road->IsSuccessor(successor_road))
+            {
+                return 1;
+            }
+            else
+            {
+                return 0;
+            }
```
Comment (oscjoha) [inline EnvironmentSimulator/Libraries/esminiRMLib/esminiRMLib.cpp]: Here I would have done like this instead but doesn't really matter, just cleaner in my opinion

```suggestion
            return (int)road->IsSuccessor(successor_road)
```

**Review by oscjoha — COMMENTED** [review]
Looks fine. However does this really work? Both branches have the source code for `IsSuccessor` commented out in the RoadManager.cpp. Double check if you missed a file and then I will approve :smiley:

**Comment by seifbourogaa** [comment]
> Looks fine. However does this really work? Both branches have the source code for `IsSuccessor` commented out in the RoadManager.cpp. Double check if you missed a file and then I will approve smiley

Correct, this does not currently work on the master branch of RISE/Esmini. I forked from the Esmini/Esmini master branch, where everything is up to date and works, but I don't have permissions to make PR on the repo without making them from this repo. 

In order to get this to work we need to have the RoadManager refactorization made by you and Adam merged into the Esmini/Esmini first, and then have this merged into Esmini/Esmini. Or, is there a better way to go about it? Please let me know. :) 

Commited your code suggestion, that's the correct way to do it of course. IsSuccessor gives us all the return values, so there's no need to double check them again. Good catch! :)

**Review by eriadam36 — APPROVED** [review]
it's alright. also think roadmanager has a get succesor maybe.

**Comment by seifbourogaa** [comment]
> it's alright. also think roadmanager has a get succesor maybe.

Correct, I am usi g it? It just wasnt open on the .so file we use. :)

**Pull request merged**